src/NHibernate/Util/SerializableConstructorInfo.cs

```diff
@@ -0,0 +1,101 @@
+		{
+			return constructorInfo == null ? null : new SerializableConstructorInfo(constructorInfo);
+		}
+
+		/// <summary>
+		/// Creates a new <see cref="SerializableConstructorInfo"/>
+		/// </summary>
+		/// <param name="constructorInfo">The <see cref="ConstructorInfo"/> being wrapped for serialization.</param>
+		private SerializableConstructorInfo(ConstructorInfo constructorInfo)
+		{
+			_constructorInfo = constructorInfo ?? throw new ArgumentNullException(nameof(constructorInfo));
+			if (constructorInfo.DeclaringType == null)
+			{
+				throw new ArgumentException("ConstructorInfo must have non-null DeclaringType", nameof(constructorInfo));
+			}
+		}
+
+		private SerializableConstructorInfo(SerializationInfo info, StreamingContext context)
+		{
+			System.Type declaringType = info.GetValue<SerializableSystemType>("declaringType").GetType();
+			SerializableSystemType[] parameterSystemTypes = info.GetValue<SerializableSystemType[]>("parameterTypesHelper");
+
+			System.Type[] parameterTypes = parameterSystemTypes?.Select(x => x.GetType()).ToArray() ?? new System.Type[0];
+			_constructorInfo = declaringType.GetConstructor(BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic, null, CallingConventions.Any, parameterTypes, null);
+
+			if (_constructorInfo == null) throw new MissingMethodException(declaringType.FullName, ".ctor");
+		}
+
+		[SecurityCritical]
+		public void GetObjectData(SerializationInfo info, StreamingContext context)
+		{
+			SerializableSystemType[] parameterSystemTypes =
+				_constructorInfo.GetParameters()
+				                .Select(x => SerializableSystemType.Wrap(x.ParameterType))
+				                .ToArray();
+
+			info.AddValue("declaringType", SerializableSystemType.Wrap(_constructorInfo.DeclaringType));
+			info.AddValue("parameterTypesHelper", parameterSystemTypes);
+		}
+
+		public ConstructorInfo Value => _constructorInfo;
+
+		private bool Equals(SerializableConstructorInfo other)
+		{
+			return Equals(_constructorInfo, other._constructorInfo);
+		}
+
+		public override bool Equals(object obj)
+		{
+			if (ReferenceEquals(null, obj)) return false;
+			if (ReferenceEquals(this, obj)) return true;
+			return obj is SerializableConstructorInfo && Equals((SerializableConstructorInfo) obj);
+		}
+
+		public override int GetHashCode()
+		{
+			return (_constructorInfo != null ? _constructorInfo.GetHashCode() : 0);
+		}
+
+		public static bool operator ==(SerializableConstructorInfo left, SerializableConstructorInfo right)
+		{
+			return Equals(left, right);
+		}
+
+		public static bool operator !=(SerializableConstructorInfo left, SerializableConstructorInfo right)
+		{
+			return !Equals(left, right);
+		}
+
+		public static implicit operator ConstructorInfo(SerializableConstructorInfo serializableConstructorInfo)
+		{
+			return serializableConstructorInfo?.Value;
+		}
+
+		public static explicit operator SerializableConstructorInfo(ConstructorInfo constructorInfo)
```

Comment:
Why is this one explicit? That is the other way round in `SerializableSystemType`, and it looks that for `SerializableConstructorInfo` the conversion is safe in both sens.

Same question for `SerializableFieldInfo`, `SerializableMethodInfo` and `SerializablePropertyInfo`.

(I am not changing that in my PR on your branch.)

Reply:
The explicit is because the conversion could have thrown.  I've removed all the operators.